alpha/veneer/basic/basic.go

```diff
@@ -0,0 +1,83 @@
+package basic
+
+import (
+	"context"
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/operator-framework/operator-registry/alpha/action"
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"github.com/operator-framework/operator-registry/pkg/image"
+)
+
+type BasicVeneer struct {
+	Registry image.Registry
+}
+
+func (v BasicVeneer) Ingest(ctx context.Context, ref string) (*declcfg.DeclarativeConfig, error) {
+	// only taking first argument as file
+	stat, serr := os.Stat(ref)
+	if serr != nil {
+		return nil, serr
+	}
+
+	if stat.IsDir() {
+		return nil, errors.New("cannot ingest veneers by directory reference") // BUGBUG enhance to walk and eat all files in turn, heeding ignore file
+	}
+	return v.renderFile(ctx, ref)
+}
+
+func (v BasicVeneer) renderFile(ctx context.Context, ref string) (*declcfg.DeclarativeConfig, error) {
+	var cfg = &declcfg.DeclarativeConfig{}
+
+	// xform any relative to absolute paths
+	abspath, err := filepath.Abs(ref)
+	if err != nil {
+		return nil, err
+	}
+	// xform to break apart dir/file elements
+	rpath, fname := filepath.Split(abspath)
+
+	root := os.DirFS(rpath)
+	file, err := root.Open(fname)
+	if err != nil {
+		return nil, err
+	}
+
+	defer file.Close()
+
+	if cfg, err = declcfg.LoadFile(root, file, ref); err != nil {
+		return cfg, err
+	}
+
+	var outb = cfg.Bundles[:0] // allocate based on max size of input, but empty slice
+	var r action.Render
+	// populate registry, incl any flags from CLI
+	if v.Registry != nil {
+		r.Registry = v.Registry
+	}
+
+	for _, b := range cfg.Bundles {
+		if isBundleVeneer(&b) {
+			r.Refs = []string{b.Image}
+			contributor, err := r.Run(ctx)
+			if err != nil {
+				// abort processing
+				return cfg, err
```

Comment:
Nit: Always return `nil` for the first return value when errors occur unless there's a good reason not to.
```suggestion
				return nil, err
```

Reply:
Before the big refactoring yesterday, I had the code structured to accumulate between loop iterations, so we might have partial good results to report.  I figured that giving partial results and the error allowed the caller to determine what they wanted to do with the results. 

Now, of course all this is moot, because we only accumulate on successful iteration over ALL bundles.  

... long way of saying "I missed it in the refactoring / will fix". :S